src/main/java/picard/sam/markduplicates/util/AbstractMarkDuplicatesCommandLineProgram.java

```diff
@@ -264,6 +273,8 @@ public static void trackOpticalDuplicates(List<? extends ReadEnds> ends,
             // Variables used for optical duplicate detection and tracking
             final List<ReadEnds> trackOpticalDuplicatesF = new ArrayList<>();
             final List<ReadEnds> trackOpticalDuplicatesR = new ArrayList<>();
+            final int nOpoticalDupF;
```

Comment:
Not necessary to declare these at the top of the block, it would be clearer to declare when they're assigned to below.